python/ray/dag/compiled_dag_node.py

```diff
@@ -958,23 +994,34 @@ def _preprocess(self) -> None:
         # custom NCCL groups for a set of actors, only one is cached.
         actors_to_nccl_group_id: Dict[FrozenSet["ray.actor.ActorHandle"], str] = {}
 
-        # If a custom NCCL group is specified for P2P actors, initialize and cache
-        # the NCCL group ID.
-        if nccl_actors_p2p and self._custom_nccl_group_p2p:
-            if not set(nccl_actors_p2p).issubset(
-                set(self._custom_nccl_group_p2p.get_actor_handles())
-            ):
+        # If a custom NCCL group is specified for P2P actors, initialize and
+        # cache the NCCL group ID.
+        for (
+            custom_nccl_group,
+            nodes,
+        ) in custom_nccl_group_to_p2p_senders_and_receivers.items():
+            nccl_actors: Set["ray.actor.ActorHandle"] = {
+                node._get_actor_handle() for node in nodes
+            }
+            if None in nccl_actors:
+                raise ValueError("Driver cannot participate in the NCCL group.")
+            if not nccl_actors.issubset(custom_nccl_group.get_actor_handles()):
                 raise ValueError(
-                    "Expected P2P actor handles to be a subset of the custom NCCL group"
+                    "Custom NCCL group must contain all actors that participate "
+                    "in P2P send/recv."
```

Comment:
Can we include more info to make it error more actionable? e.g., which actors are missing, which type hint

Reply:
Included missing actors and the custom NCCL group.